src/policyengine_data/single_year_dataset.py

```diff
@@ -1 +1,157 @@
+"""
+Class for handling single-year datasets in PolicyEngine.
+"""
 
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import h5py
+import pandas as pd
+from policyengine_core.simulations import Microsimulation
+
+
+class SingleYearDataset:
+    entities: Dict[str, pd.DataFrame]
+    time_period: str
+
+    def __init__(
+        self,
+        file_path: Optional[str] = None,
+        entities: Optional[Dict[str, pd.DataFrame]] = None,
+        fiscal_year: Optional[int] = 2025,
+    ) -> None:
+        self.entities: Dict[str, pd.DataFrame] = {}
+
+        if file_path is not None:
+            self.validate_file_path(file_path)
+            with pd.HDFStore(file_path) as f:
+                self.time_period = str(f["time_period"].iloc[0])
+                # Load all entities from the file (except time_period)
+                for key in f.keys():
+                    if key != "/time_period":
+                        entity_name = key.strip("/")
+                        self.entities[entity_name] = f[entity_name]
+        else:
+            if entities is None:
+                raise ValueError(
+                    "Must provide either a file path or a dictionary of entities' dataframes."
+                )
+            self.entities = entities.copy()
+            self.time_period = str(fiscal_year)
+
+        self.data_format = "arrays"
```

Comment:
We will eventually remove this when we patch Core to accept these dataset classes